### PR TITLE
kv/RocksDBStore: add kv_stats function to print all CF info

### PR DESCRIPTION
**Pull request by wanglinke521**
- Added the function of printing all cf's rocksdb level information.
- Currently, only the default cf information is printed.
- Print all cf information to facilitate observation of the level to which rocksdb is written.
- The main purpose is to observe the usage of db/ssd.

co-author: Jrchyang Yu <yuzhiqiang_yewu@cmss.chinamobile.com>
Signed-off-by: Wang Linke <wanglinke_yewu@cmss.chinamobile.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by wanglinke521** [comment]
@ifed01 @markhpc Can you help me review this PR? Thanks.

File: src/kv/RocksDBStore.cc [hunk 1]
```diff
@@ -1436,6 +1436,29 @@ void RocksDBStore::get_statistics(Formatter *f)
       f->close_section();
     }
   }
+
+  if (cct->_conf->rocksdb_collect_all_cf_compaction_stats) {
```
Comment (ifed01) [inline src/kv/RocksDBStore.cc:1440]: IMO that's an overkill to have a separate config switch for this stuff. May be just pring all the column handles under "if (rocksdb_collect_comapction_stats)" ?

File: src/kv/RocksDBStore.cc [hunk 2]
```diff
@@ -1436,6 +1436,29 @@ void RocksDBStore::get_statistics(Formatter *f)
...
+    handles.push_back(default_cf);
+    for (auto cf : cf_handles) {
+      for (auto shard_cf : cf.second.handles) {
+        handles.push_back(shard_cf);
+      }
+    }
+    f->open_object_section("rocksdb_all_cf_statistics");
+    for (auto handle : handles) {
+      std::string stat_str;
+      bool status = db->GetProperty(handle, "rocksdb.stats", &stat_str);
```
Comment (ifed01) [inline src/kv/RocksDBStore.cc:1451]: 'status' is unused var.

Comment (wanglinke521) [reply src/kv/RocksDBStore.cc]: Fixed. If status is false, the current loop will be exited.

File: src/kv/RocksDBStore.cc (hunk 1 above)
Comment (wanglinke521) [reply src/kv/RocksDBStore.cc]: Thanks for your suggestion. Indeed, putting it in rocksdb_collect_comapction_stats is already OK.

Comment (wanglinke521) [reply src/kv/RocksDBStore.cc]: My initial concern was that someone might call this interface in our project.

**Comment by wanglinke521** [comment]
jenkins test make check

**Comment by wanglinke521** [comment]
jenkins test make check

**Comment by wanglinke521** [comment]
jenkins test make check

File: src/kv/RocksDBStore.cc [hunk 3]
```diff
@@ -1421,21 +1421,32 @@ void RocksDBStore::get_statistics(Formatter *f)
...
+      for (auto shard_cf : cf.second.handles) {
+        handles.push_back(shard_cf);
+      }
+    }
+    f->open_object_section("rocksdb_statistics");
+    for (auto handle : handles) {
+      std::string stat_str;
+      bool status = db->GetProperty(handle, "rocksdb.stats", &stat_str);
+      if (!status) {
+        continue;
```
Comment (ifed01) [inline src/kv/RocksDBStore.cc:1438]: Please log an error at this point

Comment (wanglinke521) [reply src/kv/RocksDBStore.cc:1439]: added

Comment (ifed01) [reply src/kv/RocksDBStore.cc:1439]: IMO better use derr.

Comment (ifed01) [reply src/kv/RocksDBStore.cc:1439]: And can we get readable CF name from ColumnFamilyHandle? If so - please print it in the error message too.

Comment (wanglinke521) [reply src/kv/RocksDBStore.cc:1440]: Thanks for your suggestion. Added.

**Review by ifed01 — APPROVED** [review]

**Comment by NitzanMordhai** [comment]
Rados approved: https://tracker.ceph.com/projects/rados/wiki/MAIN#httpstrackercephcomissues67266

**Pull request merged**